tests/bsim/bluetooth/mesh/src/test_cdp1.c

```diff
@@ -0,0 +1,430 @@
+								.long_item = {
+									.elem_offset = 4,
+									.mod_item_idx = 2,
+								}};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_cfg_srv_mod = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_cfg_cli_mod = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_mod1 = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_mod2 = {
+	.cor_present = 1,
+	.format = 0,
+	.ext_item_cnt = 1,
+	.cor_id = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_mod3 = {
+	.cor_present = 1,
+	.format = 0,
+	.ext_item_cnt = 1,
+	.cor_id = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_mod4 = {
+	.cor_present = 1,
+	.format = 0,
+	.ext_item_cnt = 0,
+	.cor_id = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_mod5 = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_mod6 = {
+	.cor_present = 0,
+	.format = 1,
+	.ext_item_cnt = 1,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_vnd1 = {
+	.cor_present = 1,
+	.format = 0,
+	.ext_item_cnt = 1,
+	.cor_id = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem0_models[] = {
+	test_p1_cfg_srv_mod, test_p1_cfg_cli_mod, test_p1_mod1,
+	test_p1_mod2,        test_p1_mod3,        test_p1_vnd1,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem1_models[] = {
+	test_p1_mod4,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem2_models[] = {
+	test_p1_mod5,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem3_models[] = {
+	test_p1_mod5,
+};
+
+static const struct bt_mesh_comp_p1_model_item test_p1_elem4_models[] = {
+	test_p1_mod6,
+};
+
+static const struct bt_mesh_comp_p1_model_item *test_p1_elem_models[] = {
+	test_p1_elem0_models, test_p1_elem1_models, test_p1_elem2_models,
+	test_p1_elem3_models, test_p1_elem4_models,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem0 = {
+	.nsig = 5,
+	.nvnd = 1,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem1 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem2 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem3 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem test_p1_elem4 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static struct bt_mesh_comp_p1_elem test_p1_elems[] = {
+	test_p1_elem0, test_p1_elem1, test_p1_elem2, test_p1_elem3, test_p1_elem4,
+};
+
+static void provision_and_configure(struct bt_mesh_test_cfg cfg)
+{
+	int err;
+	uint8_t status;
+
+	err = bt_mesh_provision(net_key, 0, 0, 0, cfg.addr, cfg.dev_key);
+	if (err) {
+		FAIL("Provisioning failed (err %d)", err);
+		return;
```

Comment:
do not need `return` if there is `FAIL`. Here and below.

Reply:
Removed!